src/controllers/controller.h

```diff
@@ -16,7 +18,7 @@ class Controller : public QObject {
     Q_OBJECT
   public:
     explicit Controller(const QString& deviceName);
```

Comment:
I think it makes sense since this is computing data that it supposed to be send through midi. Isn't it adding complexity to add yet another interface in the global object; in particular an interface with only one function?

Reply:
The `engine` class is already there, and used for nearly every function in the mapping API other than raw IO. There is no overhead, just a different existing class.

Reply:
It makes less sense to me but I won't fight on this and move to the `engine` object.